react-icon-pack/src/Icon.tsx

```diff
@@ -1,5 +1,25 @@
-import { type IconComponent } from '../../types';
+import React from 'react';
+import * as BrokenIcons from '../dist/icons/broken';
+import * as OutlineIcons from '../dist/icons/outline';
+import * as CurvedIcons from '../dist/icons/curved';
+import { convertToPascalCase } from './utils';
+import { IconProps } from '../../types';
 
-export const Icon: IconComponent = () => {
-  return <div>Test web component</div>
+const iconVariants = {
+  broken: BrokenIcons,
+  outline: OutlineIcons,
+  curved: CurvedIcons,
+};
+
+type IconComponentName = keyof (typeof BrokenIcons | typeof OutlineIcons | typeof CurvedIcons);
+
+export default function Icon({ name, type = 'outline', color, size = 24 }: IconProps) {
```

Comment:
Can we use named exports instead of default exports? Just for consistency between app and web. Also what's the default color value on web? On mobile it's `black`

Reply:
On web it's inherited and I think it should stay this way